docs: add linear svc tutorial

robinstraub opened the pull request.

robinstraub commented:
Thank you for your review. I was having a hard time executing the code without VirtualLib (especially the cells that need compilation) as I am running it on an ARM architecture (Apple M2). To my understanding a new release is coming that should fix this issue so I'm waiting for it to apply your suggestions.

aquint-zama commented:
concrete-ml v1.0.0 with Apple Silicon support is out

kcelia commented:
Looking forward to the changes. Please keep me updated.

robinstraub commented:
Hi @kcelia, I'm working on the changes right now. About your request to change the dataset, I was wondering if the issue was the dataset (iris) being not a "real use-case", or it being used in another tutorial (I did not find any in the [docs](https://docs.zama.ai/concrete-ml/getting-started/showcase)) ?

I found the [pulsar star dataset](https://www.kaggle.com/datasets/colearninglounge/predicting-pulsar-starintermediate) does it sound more relevant to you and is it not being used in another tutorial ?

kcelia commented:
Hi @robinstraub, 

We would prefer a real use-case dataset, and the Pulsar Start dataset looks great. 
We haven't used it before, so feel free to use it.

Thanks.

robinstraub commented:
Hello @kcelia, I 
- changed the use case to the pulsar star as we talked,
- shortened the text and avoid repetitions
- added visualizations
- added a grid search to select common parameters alongside concrete-ml n_bits,
- updated the code where there was breaking changes so it is compliant with v1

Now that v1 is out, I feel the section about VirtualLib is less relevant, before that compilation/execution over Docker was taking over an hour for a simple dataset (iris) whereas now it runs in the matter of seconds. VirtualLib still works and is faster though so I kept it because it was an important part of the previous iteration.

Looking forward to your feedback.

robinstraub commented:
Hi @kcelia,

Thank you for your review and the explanations on how to use the simulation. It makes more sense to me now. I thought about making modifications to the tutorial to better emphasize this (e.g. swap the "simulation" and "hyperparameter selection" steps), but I preferred to remain true to your documentation and the [model development steps](https://docs.zama.ai/concrete-ml/getting-started/concepts#i.-model-development).

I:  
- adopted a more neutral tone, avoiding the use of "let's", future tense
- applied your reformulation suggestions
- displayed the parameters in a grid and highlighted the best choice (n_bits=3, as you mentionned)
- as well as other minor improvements you pointed out (unnecessary cell, typos...)

kcelia commented:
Hi @robinstraub,

Thanks for your changes. Just a few more comments and you're done.

- Remove this outdated comment in cell 1: `# import the iris dataset from sklearn, as well as some utilities and the LinearSVC for reference`

fd0r commented:
Hello @robinstraub , thanks a lot for your contribution!

Once you addressed the latest comments, could you open a PR with the notebook in the folder docs/advanced_examples on the branch https://github.com/zama-ai/concrete-ml/tree/tutorials please?
Using [CapitalizedWords](https://en.wikipedia.org/wiki/Camel_case) for the name of the notebook.

Once the PR is merged in the branch with the other bounty contributions we will procede to merge it on main and your work will be featured in the next release!

robinstraub commented:
Sorry @kcelia I missed your comment.

What do you mean by "remove the cell that contains this cell", you mean all of the simulation step ?

What do you mean by "please review your titles", do you refer to the notebook title, such as @fd0r  requested ?

jfrery commented:
Hi @robinstraub, @kcelia is off for some time. We are good to continue with this version of the notebook. Could you follow @fd0r steps to open a PR on concrete-ml repo so that we can integrate your work?

robinstraub commented:
@jfrery thank you, I opened a PR (https://github.com/zama-ai/concrete-ml/pull/75)
